data/wsl/UpdateInstall.ps1

```diff
@@ -0,0 +1,80 @@
+# Define a function to log messages with timestamps
+function LogMessage {
+    param (
+        [string]$Message,
+        [string]$Color = "White"
+    )
+    $Timestamp = (Get-Date).ToString("yyyy-MM-dd HH:mm:ss")
+    Write-Host "$Timestamp - $Message" -ForegroundColor $Color
+}
+
+# Start logging
+LogMessage "Starting Windows Update process..."
+
+# Create Update Session
+$Session = New-Object -ComObject Microsoft.Update.Session
+$Searcher = $Session.CreateUpdateSearcher()
+
+# Search for Updates
+LogMessage "Searching for updates..."
+$SearchResult = $Searcher.Search("IsInstalled=0")
+
+# Check if updates are available
+if ($SearchResult.Updates.Count -eq 0) {
+    LogMessage "No updates found." -Color "Yellow"
```

Comment:
```suggestion
    LogMessage "System already up-to-date" -Color "Green"
```
Suggestion: Formulate this in a positive way, because it indicates that the system is already in the desired state (no updates needed).

Yellow might suggest a warning or an error, which is not the case here IMHO.